src/history.ts

```diff
@@ -628,10 +713,20 @@ export class HistoryStorage {
   private async convertToHistory(memo: DecryptedMemo, pending: boolean, getIsLoopback: (shieldedAddress: string) => Promise<boolean>): Promise<HistoryRecordIdx[]> {
     const txHash = memo.txHash;
     if (txHash) {
-      const txData = await this.web3.eth.getTransaction(txHash);
+      let txData;
+      try { // TODO: will be moved to the separated RPC requester
+        txData = await this.web3.eth.getTransaction(txHash);
+      } catch (err) {
+        txData = null;
+      }
```

Comment:
Maybe this way is a bit simpler
```suggestion
      const txData = await this.web3.eth.getTransaction(txHash).catch(() => null);
```

Reply:
Great! Looks much more compact. Also I fix that in the next request.

Fixed in https://github.com/zkBob/zkbob-client-js/pull/97/commits/8c88d1efaadbea8678f406c37da7878d08bd957e